core/trino-main/src/main/java/io/trino/util/DateTimeUtils.java

```diff
@@ -423,22 +421,13 @@ public int parseInto(ReadWritablePeriod period, String text, int position, Local
                     }
                 }
                 else if (parsePos < 0) {
-                    parsePos = ~parsePos;
-                    if (parsePos > bestInvalidPos) {
-                        bestInvalidPos = parsePos;
-                    }
-                }
-            }
-
-            if (bestValidPos > position || (bestValidPos == position)) {
-                // Restore the state to the best valid parse.
-                if (bestValidPeriod != null) {
-                    period.setPeriod(bestValidPeriod);
+                    return ~parsePos;
```

Comment:
I think there is no reason to continue trying to parse when even one of parsers fails in this use case. It would be better to return earlier.

Reply:
In that case, I wonder how it should work when any parsers fail. Just throwing away error position and returning the longest succeeded position (or the initial position if there are no successful parsers) is expected?

Sure, that works because `PeriodFormatter` throws an exception when returned position is less than the text length: https://github.com/JodaOrg/joda-time/blob/v2.12.7/src/main/java/org/joda/time/format/PeriodFormatter.java#L323-L331